src/metabase/query_processor/util.clj

```diff
@@ -114,9 +115,13 @@
   ^bytes [query :- [:maybe :map]]
   ;; convert to pMBQL first if this is a legacy query.
   (let [query (try
-                (cond-> query
-                  (#{"query" "native"} (:type query)) (#(lib.convert/->pMBQL (mbql.normalize/normalize %)))
-                  (#{:query :native} (:type query))   lib.convert/->pMBQL)
+                ;; Expression type check supression is necessary because coerced fields in `query` may not have
+                ;; `:effective-type` populated. That's the case during call to this function in
+                ;; `process-userland-query-middleware` that occurs before normalization.
+                (binding [lib.schema.expression/*suppress-expression-type-check?* true]
+                  (cond-> query
+                    (#{"query" "native"} (:type query)) (#(lib.convert/->pMBQL (mbql.normalize/normalize %)))
+                    (#{:query :native} (:type query))   lib.convert/->pMBQL))
```

Comment:
This suppresses BE warning, but looking back, I believe the warning should not be suppressed. We should rather ensure that `->pMBQL` is called on correct queries (eg. not containing filter expressions that use datetime functions on coerced fields containing only base type and no effective type).

Reply:
I'd like to consult this on weekly before merging.